### DISCUSSION TITLE
How would you set route following two sets of similar routes to different machines?

### DISCUSSION THREAD

**Discussion by sweetodev** [ThreeMammals/Ocelot#2000] [Q&A] [answered]
How would you set re-reoute following two set of similar routes to different machines. Please advise.

Problem: The conflicting situation is between `/customers/1` & `/customers/1/products ` where each one is supposed to go to a different machine.

Their routes are configured as `/customers/{id}` and `/customers/1/products`

## - machine name: `customer`

```
GET /customers
GET /customers/1
POST /customers
PUT /customers1
DELETE /customers/1
```

## - machine name: `customerproduct`
```
GET /customers/1/products
PUT /customers/1/products
```

## ocelot.json

     {
      "ReRoutes": [

        {  
          "DownstreamPathTemplate": "/customers/{id}", 
          "DownstreamScheme": "http",
          "DownstreamHostAndPorts": [
            {
              "Host": "customer",
              "Port": 80
            }
          ],
          "UpstreamPathTemplate": "/customers/{id}",
          "UpstreamHttpMethod": [ "Get", "Post", "Put", "Delete" ]
        },

        {
          "DownstreamPathTemplate": "/customers/{id}/products",
          "DownstreamScheme": "http",
          "DownstreamHostAndPorts": [
            {
              "Host": "customerproduct",
              "Port": 80
            }
          ],
          "UpstreamPathTemplate": "/customers/{id}/products",
          "UpstreamHttpMethod": [ "Get", "Put" ]
        }

      ],

      "GlobalConfiguration": {
        "BaseUrl": "http://localhost:80"
      }

    }

**Comment by raman-m** [comment, marked as answer]
I don't understand, why did you split routes to 2 files?
Why not to have the same config file aka `ocelot.json` with merged routes in one file?

> How would you set re-reoute following two set of similar routes to different machines. Please advise.

## 1
If you manage dedicated web servers manually, then copy the same config to all machines!

## 2
Prepare Docker image with Ocelot app having the config inside and your configuration will be distributed across K8s as Docker containers. All of them will have the same `ocelot.json` file for all containers.
Just update Docker image each time and start deployment of containers with new version.

## 3
We have [Administration](https://ocelot.readthedocs.io/en/latest/features/administration.html) feature for more complicated scenarios of administering remote machines.

Hope it helps!
